src/state_machine/signer/mod.rs

```diff
@@ -513,7 +513,7 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
     }
 
     fn nonce_request(&mut self, nonce_request: &NonceRequest) -> Result<Vec<Message>, Error> {
-        let mut rng = OsRng;
+        let mut rng = create_rng();
```

Comment:
Same here, please pass in the `rng`. This will make it necessary to also pass the `rng` in to `process()`, etc.